web/src/app/pages/all-pushes/all-pushes.component.ts

```diff
@@ -1,15 +1,38 @@
 import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { HttpClient } from '@angular/common/http';
+import { Observable, Subject } from 'rxjs';
+import { flatMap, map, shareReplay } from 'rxjs/operators';
+
+import { step189_2020 } from '../../../proto/step128_2020';
 
 @Component({
   selector: 'app-all-pushes',
   templateUrl: './all-pushes.component.html',
   styleUrls: ['./all-pushes.component.scss']
 })
-export class AllPushesComponent implements OnInit {
-
-  constructor() { }
+export class AllPushesComponent {
+  readonly pushDefName: Observable<string>;
+  readonly pushInfos: Observable<step189_2020.IPushInfo[]>;
```

Comment:
s/IPushInfo[]/PushInfo[]?

Reply:
The static API is a bit different (see [gist](https://gist.github.com/razvanm/84eb637ae8308648451554b4cb76f808#file-step189_2020-d-ts-L28)): the `stateInfo` from `PushInfo` is of type `step189_2020.PushInfo.IStateInfo[]`.

Reply:
Got it. Thanks!